### PR TITLE
[Datumaro] CLI updates

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

- Added info on unmatched dataset items in `diff` output
- Added `Rename` transform to rename dataset items by a regex
- Fixed an issue, where `Ctrl+C` led to exit without removing of output directories

### How has this been tested?

Manual test

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- datumaro/datumaro/cli/contexts/project/diff.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5778506&bid=18918585)

File: datumaro/datumaro/cli/contexts/project/diff.py [hunk 1]
```diff
@@ -83,11 +83,20 @@ def save_dataset_diff(self, extractor_a, extractor_b):
...
-            if item_a.id != item_b.id or not item_a.id or not item_b.id:
-                print("Dataset items #%s '%s' '%s' do not match" % \
-                    (i + 1, item_a.id, item_b.id))
-                continue
+        ids_a = set((item.id, item.subset) for item in extractor_a)
+        ids_b = set((item.id, item.subset) for item in extractor_b)
+        ids = ids_a & ids_b
+
+        if len(ids) != len(ids_a):
+            print("Unmatched items in the first dataset: ")
```
Comment (nmanovic) [inline datumaro/datumaro/cli/contexts/project/diff.py:91]: Why print instead of logs?

File: datumaro/datumaro/plugins/transforms.py [hunk 2]
```diff
@@ -372,6 +373,40 @@ def transform_item(self, item):
...
+            help="Regex for renaming.")
+        return parser
+
+    def __init__(self, extractor, regex):
+        super().__init__(extractor)
+
+        assert regex and isinstance(regex, str)
+        parts = regex.split(regex[0], maxsplit=3)
+        regex, sub = parts[1:3]
+        self._re = re.compile(regex)
```
Comment (nmanovic) [inline datumaro/datumaro/plugins/transforms.py:403]: Probably need to catch exceptions here and provide users a meaningful error message.

File: datumaro/datumaro/cli/contexts/project/diff.py (hunk 1 above)
Comment (zhiltsov-max) [reply datumaro/datumaro/cli/contexts/project/diff.py:91]: The whole component prints some information. This should probably be put to a `json` or something similar.

**Pull request merged**